File: mashup-recruit/src/main/java/kr/mashup/branding/ui/team/TeamController.java
package kr.mashup.branding.ui.team;

import java.util.List;
import java.util.stream.Collectors;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import io.swagger.annotations.ApiOperation;
import kr.mashup.branding.facade.team.TeamFacadeService;
import kr.mashup.branding.ui.ApiResponse;
import lombok.RequiredArgsConstructor;

@RestController
@RequestMapping("/api/v1/teams")
@RequiredArgsConstructor
public class TeamController {
    private final TeamFacadeService teamFacadeService;

    @ApiOperation("팀 목록 조회")
    @GetMapping
    public ApiResponse<List<TeamResponse>> getTeams() {

        List<TeamResponse> responses = teamFacadeService.getTeams();


        return ApiResponse.success(responses);
    }
}

File: mashup-recruit/src/main/java/kr/mashup/branding/ui/team/TeamController.kt
package kr.mashup.branding.ui.team

import io.swagger.annotations.ApiOperation
import kr.mashup.branding.facade.team.TeamFacadeService
import kr.mashup.branding.ui.ApiResponse
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1/teams")
class TeamController(
    private val teamFacadeService: TeamFacadeService
) {

    @ApiOperation("팀 목록 조회")
    @GetMapping
    fun getTeams(
        @RequestParam(defaultValue = "13", required = false) generationNumber: Int?
    ): ApiResponse<List<TeamResponse>> {
        val responses = teamFacadeService.getTeams(generationNumber)
        return ApiResponse.success(responses)
    }
}